libs/libarchfpga/src/physical_types_util.cpp

```diff
@@ -490,3 +857,494 @@ const t_port* get_port_by_pin(t_logical_block_type_ptr type, int pin) {
 
     return nullptr;
 }
+
+/* Access information related to pin classes */
+
+/** get information given class physical num **/
+
+std::tuple<const t_sub_tile*, int> get_sub_tile_from_class_physical_num(t_physical_tile_type_ptr physical_tile, int physical_class_num) {
```

Comment:
All the functions implementations don't have any comments. High-level comments on what is done in the function implementation is very useful